### PR TITLE
feat: RegExp support for *ByTestId queries

### PR DESCRIPTION
**Pull request by acatalina**
### Summary

All *AllByTestId queries support a RegExp matcher.

We are currently preparing to migrate to the the v7 of the testing library. In our tests, the lack of this feature is a massive disadvantage for our migration as we relay in this functionality quite often.

### Test plan

Tests added following preexisting tests.

### CODE REVIEW

File: src/helpers/getByAPI.js [hunk 1]
```diff
@@ -89,6 +89,16 @@ const getTextInputNodeByDisplayValue = (node, value) => {
   }
 };
 
+const getNodeByTestId = (node, testID) => {
+  try {
+    return typeof testID === 'string'
+      ? testID === node.props.testID
+      : testID.test(node.props.testID);
+  } catch (error) {
+    throw createLibraryNotSupportedError(error);
```
Comment (thymikee) [inline src/helpers/getByAPI.js:98]: We only use `createLibraryNotSupportedError ` when importing something from `react-native`. No need for this try/catch here

**Review by thymikee — CHANGES_REQUESTED** [review]
Awesome! Can we add this for `getByTestId` as well? :)

File: src/helpers/getByAPI.js (hunk 1 above)
Comment (acatalina) [reply src/helpers/getByAPI.js]: oh, fair enough. Sorted

**Review by thymikee — APPROVED** [review]
Looks good 👍

**Comment by acatalina** [comment]
> Awesome! Can we add this for `getByTestId` as well? :)

ha, you got me while I was updating the other *ByTestId and the docs

File: src/__tests__/getByApi.test.js [hunk 2]
```diff
@@ -35,3 +35,17 @@ test('getByTestId returns only native elements', () => {
...
+    <View>
+      <Text testID="text">Text</Text>
+      <TextInput testID="textInput" />
+      <View testID="view" />
+      <Button testID="button" title="Button" onPress={jest.fn()} />
+      <MyComponent testID="myComponent" />
+    </View>
+  );
+
+  expect(getAllByTestId(/text/)).toHaveLength(2);
```
Comment (thymikee) [inline src/__tests__/getByApi.test.js:50]: let's add a test for `getByTestId ` as well, for completeness

Comment (acatalina) [reply src/__tests__/getByApi.test.js:51]: sure, sorted

**Review by mdjastrzebski — APPROVED** [review]
LGTM

**Pull request merged**